strings/decimal.c

```diff
@@ -211,6 +211,74 @@ static const dec1 frac_max[DIG_PER_DEC1-1]={
           (to)=a;                                                       \
         } while(0)
 
+
+/*
+  This is a direct loop unrolling of code that used to look like this:
+  for (; *buf_beg < powers10[i--]; start++) ;
+
+  @param   i    start index
+  @param   val  value to compare against list of powers of 10
+
+  @retval  Number of leading zeroes that can be removed from fraction.
+
+  @note Why unroll? To get rid of lots of compiler warnings [-Warray-bounds]
+        Nice bonus: unrolled code is significantly faster.
+ */
+static inline int count_leading_zeroes(int i, dec1 val)
```

Comment:
If called with `0` for the second parameter `count_leading_zeroes(XXX, 0)` will always assert